src/shared/utils/prismLanguageMapper.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Maps file extension names to prism supported languages.
+ * The full list can be found in the prism-react-renderer package:
+ * https://github.com/FormidableLabs/prism-react-renderer/blob/v1.3.5/src/vendor/prism/includeLangs.js
+ */
+
+import { type Language } from 'prism-react-renderer'
+
+const prismSupportedLanguages = new Map<string, Language>([
+  ['html', 'markup'],
+  ['xml', 'markup'],
+  ['svg', 'markup'],
+  ['mathml', 'markup'],
+  ['ssml', 'markup'],
+  ['atom', 'markup'],
+  ['rss', 'markup'],
+  ['sh', 'bash'],
+  ['c', 'c'],
+  ['h', 'clike'],
+  ['cc', 'cpp'],
+  ['cpp', 'cpp'],
+  ['css', 'css'],
+  ['js', 'javascript'],
+  ['cjs', 'javascript'],
+  ['mjs', 'javascript'],
+  ['jsx', 'jsx'],
+  ['go', 'go'],
+  ['gql', 'graphql'],
+  ['graphql', 'graphql'],
+  ['json', 'json'],
+  ['less', 'less'],
+  ['objc', 'objectivec'],
+  ['ocaml', 'ocaml'],
+  ['py', 'python'],
+  ['reason', 'reason'],
+  ['sass', 'sass'],
+  ['scss', 'scss'],
+  ['sql', 'sql'],
+  ['ts', 'typescript'],
+  ['tsx', 'tsx'],
+  ['wasm', 'wasm'],
+  ['yaml', 'yaml'],
+])
+
+export function prismLanguageMapper(fileName: string): Language {
+  const fileExtension = fileName.split('.').pop() ?? ''
+
+  // we have to return something or else TS will complain
+  return prismSupportedLanguages.get(fileExtension) ?? 'markup'
+}
```

Comment:
It will rightfully complain though, as there is no guarantee that .get will not return undefined. If you want, something like this might make this a bit more readable and explain the intent here

```suggestion
const DEFAULT_LANGUAGE_TYPE: Language = 'markdown'
export function prismLanguageMapper(fileName: string): Language {
  const fileExtension = fileName.split('.').pop() ?? ''

  // we have to return something or else TS will complain
  return prismSupportedLanguages.get(fileExtension) ?? DEFAULT_LANGUAGE_TYPE
}
```

Reply:
Something that just occurred to me here is that we dont have any visibility into how often the default case will be hit in prod so it would be useful to captureMessage and monitor this to see if there is missing support for some popular languages folks use on codecov

```ts
const supportedLanguage = prismSupportedLanguages.get(fileExtension)
if(supportedLanguage) return supportedLanguage

Sentry.captureMessage('Unsupported language type for filename ${fileName}') // make sure to add a fingerprint here btw
return DEFAULT_LANGUAGE_TYPE
```